### PR TITLE
feat: EXPOSED-78 Support database-generated values for columns

### PR DESCRIPTION
**Pull request by joc-a**
This feature will enable marking a column as `databaseGenerated` to be able to omit setting it when inserting a new record without getting an error. The value for the column can be set with TRIGGER or DEFAULT, for example.

### CODE REVIEW

File: exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Column.kt [hunk 1]
```diff
@@ -39,6 +39,8 @@ class Column<T>(
 
     fun defaultValueInDb() = dbDefaultValue
 
+    internal var isGeneratedInDb: Boolean = false
```
Comment (e5l) [inline exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Column.kt:42]: Could we check what names are used in other frameworks for that?

**Review by e5l — APPROVED** [review]

File: exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Column.kt (hunk 1 above)
Comment (joc-a) [reply exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Column.kt:42]: Here is what I found:
- [Entity Framework - [DatabaseGenerated(DatabaseGeneratedOption.None)]](https://www.learnentityframeworkcore.com/configuration/data-annotation-attributes/databasegenerated-attribute)
- [Hibernate - @Generated](https://docs.jboss.org/hibernate/orm/6.2/javadocs/org/hibernate/annotations/Generated.html)

I didn't find this feature in Ktorm and jOOQ.

File: exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/entities/EntityTests.kt [hunk 2]
```diff
@@ -1400,4 +1400,76 @@ class EntityTests : DatabaseTestsBase() {
...
+        companion object : IntEntityClass<CreditCard>(CreditCards)
+
+        var number by CreditCards.number
+        var spendingLimit by CreditCards.spendingLimit
+    }
+
+    @Test
+    fun testDbGeneratedDefault() {
+        withTables(excludeSettings = listOf(TestDB.SQLITE), CreditCards) { testDb ->
+            addLogger(StdOutSqlLogger)
```
Comment (bog-walk) [inline exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/entities/EntityTests.kt:1419]: Just noting that a logger got accidentally left in.

**Review by bog-walk — APPROVED** [review]

File: exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Column.kt (hunk 1 above)
Comment (e5l) [reply exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Column.kt]: `databaseGenerated` lgtm

**Pull request merged**